transaction.go

```diff
@@ -480,31 +480,16 @@ func TxSemanticInputUnlocks() TxSemanticValidationFunc {
 				if chainID.Empty() {
 					chainID = chainID.(UTXOIDChainID).FromOutputID(svCtx.WorkingSet.UTXOInputAtIndex(uint16(inputIndex)).Ref())
 				}
-				svCtx.WorkingSet.UnlockedIdents.AddUnlockedChain(chainID.ToAddress(), uint16(inputIndex))
-			}
-		}
 
-		// check whether any alias output which is referenced is not doing a state transition
-		for _, v := range svCtx.WorkingSet.UnlockedIdents {
-			if len(v.ReferencedBy) == 0 {
-				continue
-			}
-
-			input := svCtx.WorkingSet.Inputs[v.UnlockedAt]
-			current, ok := input.(*AliasOutput)
-			if !ok {
-				continue
-			}
-
-			// there can be multiple entries given a chain output
-			aliasAddr, ok := v.Ident.(*AliasAddress)
-			if !ok {
-				continue
-			}
+				// for alias outputs which are not state transitioning, we do not add it to the set of unlocked chains
+				if current, ok := chainConstrOutput.(*AliasOutput); ok {
+					next, hasNextState := svCtx.WorkingSet.OutChains[chainID]
+					if !hasNextState || (current.StateIndex+1 != next.(*AliasOutput).StateIndex) {
```

Comment:
I was not able to do it, but do you think someone could create a malicious transaction somehow, where this chainID points to an NFTOutput instead, and cause a panic in the lib?

Reply:
Funny, I think I created a test case where I have the exact same key twice in the OutChains map, but go still picks the right result from the map because one key seems to be an `aliasID` and the other one an `nftID`.